### PR TITLE
fix(path_distance_calculator): do not calculate distance when too short path is subscribed

### PR DESCRIPTION
**Pull request by tkimura4**
## Description
Do not calculate distance when too short path is subscribed to avoid process-death by invalid access here
```
    const double distance =
      motion_utils::calcSignedArcLength(path->points, pose->pose.position, path->points.size() - 1);
```

## Related links
[TERIV Internal](https://star4.slack.com/archives/C017VB9UG1L/p1676457271165259)

## Tests performed
I confirmed that the path_distance_calculator node works normally using the planning simulator.
$ros2 launch path_distance_calculator path_distance_calculator.launch.xml 

(But I didn't test that the node didn't die when a too short path is subscribed.)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by tkimura4** [comment]
@yabuta @tkhmy 
Could you review this?

**Comment by tkimura4** [comment]
@isamu-takagi  @yabuta 
Could you review this?

File: common/path_distance_calculator/src/path_distance_calculator.cpp [hunk 1]
```diff
@@ -46,10 +46,12 @@ PathDistanceCalculator::PathDistanceCalculator(const rclcpp::NodeOptions & optio
     }
     if (path->points.empty()) {
       RCLCPP_WARN_THROTTLE(get_logger(), *get_clock(), 5000, "path empty");
+      return;
     }
 
     if (path->points.size() == 1) {
       RCLCPP_WARN_THROTTLE(get_logger(), *get_clock(), 5000, "too short path");
+      return;
     }
```
Comment (isamu-takagi) [inline common/path_distance_calculator/src/path_distance_calculator.cpp:55]: These can be merged into `path->points.size() <= 1`.

Comment (tkimura4) [reply common/path_distance_calculator/src/path_distance_calculator.cpp]: Thansk.
addressed in [85a3ab8](https://github.com/autowarefoundation/autoware.universe/pull/2892/commits/85a3ab85d26b14a01e852d75ac065690e8474a00)

**Review by isamu-takagi — APPROVED** [review]
LGTM. I leave it up to you to reflect the comments.

**Pull request merged**